ouroboros-consensus/src/Ouroboros/Consensus/Protocol/PBFT/ChainState.hs

```diff
@@ -127,10 +129,41 @@ data PBftChainState c = PBftChainState {
       --
       -- We should have precisely @n@ signatures in the window, unless we are
       -- near genesis.
+      --
+      -- INVARIANT Empty if and only if we are exactly at genesis.
```

Comment:
I think this invariant already held -- I didn't add anything to ensure it. I just added the comment to explicit list it.